webpack/components/PreupgradeReports/PreupgradeReportsHelpers.js

```diff
@@ -0,0 +1,30 @@
+export const flattenEntries = reports =>
+  reports.reduce((memo, report) => [...memo, ...report.entries], []);
+
+export const entryFixable = entry =>
+  entry.detail &&
+  entry.detail.remediations &&
+  entry.detail.remediations.some(remediation => remediation.type === 'command');
+
+export const isEmpty = obj => Object.keys(obj).length === 0;
+
+export const anyEntriesFixable = reports =>
+  flattenEntries(reports).some(entryFixable);
+
+export const idsForInvocation = reports =>
+  reports.reduce(
+    (memo, report) => {
+      report.entries.map(entry => {
+        if (entryFixable(entry)) {
+          memo.entryIds = [...memo.entryIds, entry.id];
+
+          if (!memo.hostIds.includes(report.hostId)) {
+            memo.hostIds = [...memo.hostIds, report.hostId];
+          }
+        }
+        return entry;
+      });
```

Comment:
If I'm reading this right, it iterates through all entries, modifying `memo.entryIds` and `memo.hostIds` as it goes. And then the result of this is discarded and only the "side-effects" (modification of memo) are used. If that is the case, wouldn't it be better if this used `each` instead of `map` and dropped the `return entry` at the end?

Reply:
No longer using `map`, hopefully the intention is more clear now.